EDU-3304 - Document next retry delay in .NET (#3085)
temporalio/documentation: docs/develop/dotnet/failure-detection.mdx

    });
```

> ### Overriding the retry interval with Next Retry Delay {#next-retry-delay}

Reviewer: ```suggestion
Override the Retry interval with `nextRetryDelay` {#next-retry-delay}
```

    });
```

> Override the Retry interval with `nextRetryDelay` {#next-retry-delay}

Reviewer: ```suggestion
### Override the Retry interval with `nextRetryDelay` {#next-retry-delay}
```

    });
```

> ### Overriding the retry interval with Next Retry Delay {#next-retry-delay}

Reviewer: ```suggestion
### Override the retry interval with `NextRetryDelay` {#next-retry-delay}
```

- We use infinitive/base form, not present participle.
- When using specific code-derived terms, such as field names, use code face without spaces.

Author: This is what is already present at `docs/develop/java/failure-detection.mdx` and may be a rule worth applying consistently there too instead of just here (but feel free to apply here first)

> You may throw an [Application Failure](/references/failures#application-failure) with the NextRetryDelay field set. This value will replace and override whatever the retry interval would be on the retry policy.

Reviewer: ```suggestion
Throw an [Application Failure](/references/failures#application-failure) with an assigned `NextRetryDelay` field to replace and override the Retry interval defined in the Retry Policy.
```

Author: This is the form already present/approved at `docs/develop/java/failure-detection.mdx` and may be a rule worth applying consistently there too instead of just here (but feel free to apply here first)

> For example, if in an activity, you want to base the interval on the number of attempts, you might do:

Reviewer: ```suggestion
For example, you might scale the next Retry delay interval based on the current number of attempts.
Here's how you'd do that in an Activity:
```

- Activity is a reserved term, so capitalized

Author: This is the form already present/approved at `docs/develop/java/failure-detection.mdx` and may be a rule worth applying consistently there too instead of just here (but feel free to apply here first)

```csharp

> var attempt = ActivityExecutionContext.Current.Info.Attempt;

Reviewer: It may be "obvious" by name and context that this is "retrieve attempt count from the Activity Execution context", but a comment wouldn't hurt, and ditto below, "Update the next Retry delay by assigning a Time value to the `nextRetryDelay` field"

Also, this uses lower case n for Next. Is the material preceding incorrect?

Author: This is the form already present/approved at `docs/develop/java/failure-detection.mdx` and may be a rule worth applying consistently there too instead of just here (but feel free to apply here first)
